src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -184,13 +223,79 @@ private List<ProgrammingExerciseTask> extractTasks(ProgrammingExercise exercise)
             var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
 
             for (String testName : testCaseNames) {
-                testCases.stream().filter(tc -> tc.getTestName().equals(testName)).findFirst().ifPresent(task.getTestCases()::add);
+                Optional<ProgrammingExerciseTestCase> foundTestCase;
+                if (testName.startsWith(TESTID_START)) {
+                    Long id = extractTestId(testName);
+                    foundTestCase = testCases.stream().filter(tc -> tc.getId().equals(id)).findFirst();
+                }
+                else {
+                    foundTestCase = testCases.stream().filter(tc -> tc.getTestName().equals(testName)).findFirst();
+                }
+                foundTestCase.ifPresent(task.getTestCases()::add);
             }
             tasks.add(task);
         }
         return tasks;
     }
 
+    /**
+     * Replaces a comma seperated list of test case names with their corresponding ids.
+     * If no matching test case exists (e.g. due to a typo), we keep the test name.
+     */
+    private String extractTestCaseIdsFromNames(String capturedTestCaseNames, Set<ProgrammingExerciseTestCase> testCases) {
+        var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
+
+        return testCaseNames.stream().map(testName -> convertTestNameToTestId(testName, testCases)).collect(Collectors.joining(","));
+    }
+
+    private String convertTestNameToTestId(String testName, Set<ProgrammingExerciseTestCase> testCases) {
+        for (ProgrammingExerciseTestCase testCase : testCases) {
+            if (testName.equals(testCase.getTestName())) {
+                String id = testCase.getId().toString();
+                return TESTID_START + id + TESTID_END;
```

Comment:
Optional note: I would not expect a method called `convertTestNameToTestId` to also add `<testid>...</testid>` around the ID 🤔 Maybe rename the method or add it somewhere else? I would expect the test ID to be the content between the `<testid>` tags and not the full thing.